feature/experimental/bgp/ate_tests/bgp_always_compare_med/bgp_always_compare_med_test.go

```diff
@@ -0,0 +1,487 @@
+	at := gnmi.OC()
+
+	rib := at.NetworkInstance(ateSrc.Name).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "0").Bgp().Rib()
+	prefixPath := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
+		NeighborAny().AdjRibInPre().RouteAny().WithPathId(0).Prefix()
+
+	gnmi.WatchAll(t, ate, prefixPath.State(), time.Minute, func(v *ygnmi.Value[string]) bool {
+		_, present := v.Val()
+		return present
+	}).Await(t)
+
+	wantMED := []uint32{}
+	// Build wantMED to compare the diff.
+	for i := 0; i < routeCount; i++ {
+		wantMED = append(wantMED, uint32(wantMEDValue))
+	}
+
+	gotMED := gnmi.GetAll(t, ate, rib.AttrSetAny().Med().State())
+	if diff := cmp.Diff(wantMED, gotMED); diff != "" {
+		t.Errorf("Obtained MED on ATE is not as expected, got %v, want %v", gotMED, wantMED)
+	}
+}
+
+// verifyBGPCapabilities is used to Verify BGP capabilities like route refresh as32 and mpbgp.
+func verifyBGPCapabilities(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Log("Verifying BGP capabilities.")
+	var nbrIP = []string{ateSrc.IPv4, ateDst1.IPv4, ateDst2.IPv4}
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	for _, nbr := range nbrIP {
+		nbrPath := statePath.Neighbor(nbr)
+		capabilities := map[oc.E_BgpTypes_BGP_CAPABILITY]bool{
+			oc.BgpTypes_BGP_CAPABILITY_ROUTE_REFRESH: false,
+			oc.BgpTypes_BGP_CAPABILITY_MPBGP:         false,
+		}
+		for _, cap := range gnmi.Get(t, dut, nbrPath.SupportedCapabilities().State()) {
+			capabilities[cap] = true
+		}
+		for cap, present := range capabilities {
+			if !present {
+				t.Errorf("Capability not reported: %v", cap)
+			}
+		}
+	}
+}
+
+// verifyPrefixesTelemetry confirms that the dut shows the correct numbers of installed,
+// sent and received IPv4 prefixes.
+func verifyPrefixesTelemetry(t *testing.T, dut *ondatra.DUTDevice, wantInstalled, wantSent uint32) {
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	prefixesv4 := statePath.Neighbor(ateSrc.IPv4).AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Prefixes()
+	if gotInstalled := gnmi.Get(t, dut, prefixesv4.Installed().State()); gotInstalled != wantInstalled {
+		t.Errorf("Installed prefixes mismatch: got %v, want %v", gotInstalled, wantInstalled)
+	}
+	if gotSent := gnmi.Get(t, dut, prefixesv4.Sent().State()); gotSent != wantSent {
+		t.Errorf("Sent prefixes mismatch: got %v, want %v", gotSent, wantSent)
+	}
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+	peerGrp    string
+}
+
+// TestRemovePrivateAS is to Validate that private AS numbers are stripped
+// before advertisement to the eBGP neighbor.
+func TestAlwaysCompareMED(t *testing.T) {
+	// DUT configurations.
+	t.Logf("Start DUT config load.")
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+	d := &oc.Root{}
```

Comment:
d is a little bit of a cryptic variable name, even for Go. Maybe rename it "root" throughout?

Reply:
changed to dutOcRoot.